### DISCUSSION TITLE
Minimize quality loss and resize image down.

### DISCUSSION THREAD

**Discussion by lm-sudo** [ImageMagick/ImageMagick#6896] [Help]
### ImageMagick version

7.1.1-21

### Operating system, version and so on

Ubuntu 20.04.6 LTS

### Description

I have a png(of a label, no background) with dimensions 1123x431. I need to size it down to dimensions 195x75 with as little loss of quality as possible as there is text which must be legible and a barcode which must be scannable printed on the label+. I have tried many variations/combinations of resize and thumbnail but I can't make my label legible at 195x75.

Any recommendations for this? I have searched documentation, stack overflow, and this forum but I can't seem to find anything that works. I am also open to converting to another file type(pdf?) and then resizing if there is a file type which lends itself to resizing with less loss of quality. I am also open to using other tools (pdftk, ghostscript) if they are necessary.

Please excuse if I am incorrectly using terminology as I have only recently taken up image magick and am very green. I'll be happy to try to clarify any ambiguities in my question. I have also attached a sample label which is similar to the one I am working with.

Thank you.
![test_cropping_cropped](https://github.com/ImageMagick/ImageMagick/assets/89110072/04980de1-80ba-4379-ab41-9a3f268c3200)

**Comment by snibgo** [comment]
The problem isn't in the file type, but the amout of reduction.

You want to reduce by a factor of 1123/195 = 5.78.

The barcode has a gap between black lines of about 6 pixels, so the white gaps will reduce to about 1 pixel. They are just about legible, but not reliably scannable.

> ... as there is text which must be legible ...

No chance. A capital "L" is 19 pixels high, reduced to about 3 pixels. Lower-case "e" and "a" are about 13 pixels high, which reduce to about 2 pixels high. With that height, we cannot recognize either letter.

**Reply by lm-sudo** [reply]
I was able to near-losslessly do the conversion although it required ghostscipt, I was unable to do it through image magick. Not sure if I should post the solution I used since it isn't image magick.

**Comment by snibgo** [comment]
Yes, please show your solution.

**Reply by lm-sudo** [reply]
//take png and convert any transparency to white:
magick image4.png -size XxY xc:white +swap -compose over -composite image4.jpg

//convert jpg to pdf
img2pdf image4.jpg -o image4.pdf

//the line below converts a pdf of a label that's an arbitrary size to the dimensions
//provided in devicewidthpoints and deviceheightpoints
 gs  -o output.pdf  -sDEVICE=pdfwrite  -dDEVICEWIDTHPOINTS=195 -dDEVICEHEIGHTPOINTS=75  -dFIXEDMEDIA  -dPDFFitPage  -dCompatibilityLevel=1.4   image4.pdf

I don't know how "lossless" this is, but I know that scaling down my image to 195x75 I am able to instantly scan the label with various scanners I've tested.
